app/services/process_payment_intent.rb

```diff
@@ -3,39 +3,63 @@
 # When directing a customer to Stripe to authorize the payment, we specify a
 # redirect_url that Stripe should return them to. When checking out, it's
 # /checkout; for admin payments and subscription payemnts it's the order url.
+#
 # This class confirms that the payment intent matches what's in our database,
 # marks the payment as complete, and removes the cvv_response_message field,
 # which we use to indicate that authorization is required. It also completes the
 # Order, if appropriate.
 
 class ProcessPaymentIntent
+  class Result
+    attr_reader :error
+
+    def initialize(ok:, error: nil)
+      @ok = ok
+      @error = error
+    end
+
+    def ok?
+      @ok
+    end
+  end
+
   def initialize(payment_intent, order)
     @payment_intent = payment_intent
     @order = order
     @last_payment = OrderPaymentFinder.new(order).last_payment
   end
 
   def call!
-    return unless valid?
+    validate_intent!
+    return Result.new(ok: false, error: "") unless valid?
+
+    mark_as_processed
 
-    last_payment.update_attribute(:cvv_response_message, nil)
     OrderWorkflow.new(@order).next
     last_payment.complete! if last_payment.can_complete?
+
+    Result.new(ok: true)
+  rescue Stripe::StripeError => e
+    Result.new(ok: false, error: e.message)
   end
 
   private
 
   attr_reader :order, :payment_intent, :last_payment
 
   def valid?
-    order.present? && valid_intent_string? && matches_last_payment?
+    order.present? && matches_last_payment?
   end
 
-  def valid_intent_string?
-    payment_intent&.starts_with?("pi_")
+  def validate_intent!
+    Stripe::PaymentIntentValidator.new.call(payment_intent, "stripe_account_id")
```

Comment:
`stripe_account_id` is what I need to figure out. I need to see where to pick that and how it's supposed to work.